neo/SmartContract/Iterators/ArrayWrapper.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Neo.VM;
+using System;
+using System.Collections.Generic;
+
+namespace Neo.SmartContract.Iterators
+{
+    internal class ArrayWrapper : IIterator
+    {
+        private readonly IList<StackItem> array;
+        private int index = -1;
+
+        public ArrayWrapper(IList<StackItem> array)
+        {
+            this.array = array;
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public StackItem Key()
+        {
+            if (index < 0)
+                throw new InvalidOperationException();
+            return index;
+        }
+
+        public bool Next()
+        {
+            int next = index + 1;
+            if (next >= array.Count)
+                return false;
+            index = next;
+            return true;
+        }
```

Comment:
Is it mandatory to execute Next() before getting any element? 
I mean, if you initialize iterator and try to get element it will fail (because index == -1), only after one Next() call.

Reply:
Yes. You need to call it like this:

```c#
IIterator itor;
while (itor.Next())
{
    var key = itor.Key();
    var value = itor.Value();
    //do something here with key and value;
}
```